contrib/charts/templates/deployment-analyzer.yaml

```diff
@@ -17,56 +17,110 @@ spec:
         productVersion: {{ .Chart.Version }}
       labels:
         app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
         tier: analyzer
     spec:
+      serviceAccountName: skydive-service-account
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      hostPID: true
+      hostIPC: true
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1000
       affinity:
         {{- include "nodeaffinity" . | indent 6 }}
-{{ if .Values.image.imagePullSecrets }}
+      {{- if .Values.image.secretName }}
+      {{- if ne .Values.image.secretName ""}}
       imagePullSecrets:
-      - name: {{ .Values.image.imagePullSecrets }}
-{{ end }}
+      - name: {{ .Values.image.secretName }}
+      {{- end }}
+      {{- end }}      
       containers:
       - name: skydive-analyzer
         image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
         imagePullPolicy: {{ default "" .Values.image.imagePullPolicy | quote }}
+        terminationMessagePolicy: FallbackToLogsOnError
         args:
         - analyzer
-        - --listen=0.0.0.0:8082
+        - --listen=0.0.0.0:{{ .Values.service.port }}
         ports:
         - containerPort: {{ .Values.service.port }}
         - containerPort: {{ .Values.service.port }}
           protocol: UDP
-        - containerPort: 12379
-        - containerPort: 12380
+        - containerPort: {{ .Values.etcd.port }}
+        securityContext:
+          privileged: false
+          readOnlyRootFilesystem: false
+          allowPrivilegeEscalation: false
+          runAsNonRoot: false
+          runAsUser: 0
+          capabilities:
+                add:
+                - SYS_ADMIN
```

Comment:
Why giving these previleges to the analyzer ?

Reply:
removed